src/userInterface/panels/CommandPanel/CommandPanel.tsx

```diff
@@ -92,90 +84,115 @@ export const CommandPanel: React.FC<{
+                    save();
+                }
+            });
         }
         else {
             confirm({
                 title: LOCALIZATION.SAVE_TITLE,
                 icon: <ExclamationCircleOutlined />,
                 content: LOCALIZATION.SAVE_CONTENT,
                 onOk() {
-                    props.commandEvent(CommandEvent.saveAs)
+                    save();
+                }
+            });
+        }
+    }
+
+    const update = async () => {
+        if (await isUpdated(props.selectedJsonConfigId)) {
+            props.setMergeOptions({
+                localConfig: JsonConfigCommandCenter.currentJson,
+                serverConfig: await JsonConfigCommandCenter.loadJsonConfigs()
+                    .then(response => {
+                        if (response) {
+                            return response.get(props.selectedJsonConfigId).data;
+                        }
+                    })
+                    .catch(error => {
+                        message.error(LOCALIZATION.RETRIEVE_CONFIGS_FAIL.replace('{{error}}', `${extractErrorMessage(error)}`));
+                    }),
+                onOk: (mergedJson: any) => {
+                    props.commandEvent(CommandEvent.update, mergedJson)
                         .then((response: any) => {
                             const createdId = response.data.data.items[0].id;
                             props.reloadJsonConfigs(createdId);
                             JsonConfigCommandCenter.updateTitle();
-                            message.success(LOCALIZATION.SAVE_SUCCESS);
+                            message.success(LOCALIZATION.UPLOAD_SUCCESS);
                         })
                         .catch((error: any) => {
-                            message.error(LOCALIZATION.SAVE_ERROR.replace('{{error}}', `${extractErrorMessage(error)}`));
+                            message.error(LOCALIZATION.UPLOAD_ERROR.replace('{{error}}', `${extractErrorMessage(error)}`));
                         });
+                },
+                onCancel: () => {
+                    message.warning(LOCALIZATION.UPLOAD_ERROR.replace('{{error}}', ''));
                 }
             });
         }
+        else {
+            props.commandEvent(CommandEvent.update)
+                .then((response: any) => {
+                    const createdId = response.data.data.items[0].id;
+                    props.reloadJsonConfigs(createdId);
+                    JsonConfigCommandCenter.updateTitle();
+                    message.success(LOCALIZATION.UPLOAD_SUCCESS);
+                })
+                .catch((error: any) => {
+                    message.error(LOCALIZATION.UPLOAD_ERROR.replace('{{error}}', `${extractErrorMessage(error)}`));
+                });
+        }
     }
 
     const onUpdateHandler = () => {
-        if (!isValidFileName()) {
-            message.error(LOCALIZATION.UPLOAD_INVALID_FILE);
+        if (JsonConfigCommandCenter.hasErrors()) {
```

Comment:
why cant hasError be a boolean property?

Reply:
changed